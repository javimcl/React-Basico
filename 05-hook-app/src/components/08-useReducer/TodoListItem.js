import React from 'react';

export const TodoListItem = ({todo, i, handleToggle, handleDelete }) => {
    return (
        <>
            <li key={todo.id}
                className="list-group-item"
            >
                <p className={`${todo.done && 'complete'}`}
                    onClick={() => handleToggle(todo.id)}>
                    {i + 1}. {todo.desc}
                </p>
                <button

                    onClick={() => handleDelete(todo.id)}
                    className='btn btn-danger'>
                    Borrar
                </button>
            </li>
        </>
    );
};
