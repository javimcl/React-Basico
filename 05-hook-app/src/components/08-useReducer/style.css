body {
    padding: 70px;
}

p {
    cursor: pointer;
    margin: opx;

}

.list-group-item {
    align-items: center;
    display: flex;
    justify-content: space-between;
}

.complete {
    text-decoration: line-through;
}