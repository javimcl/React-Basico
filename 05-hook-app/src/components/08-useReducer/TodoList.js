import React from 'react';
import { TodoListItem } from './TodoListItem';

export const TodoList = ({todos, handleDelete, handleToggle}) => {
    return (
        <div>
            
            <ui className="list-group list-group-flush">
                        {
                            todos.map((todo, i) => (
                                <TodoListItem key={todo.id} todo={todo} i={i} handleDelete={handleDelete} handleToggle={handleToggle} />
                            ))
                        }
                    </ui>
            
        </div>
    );
};
