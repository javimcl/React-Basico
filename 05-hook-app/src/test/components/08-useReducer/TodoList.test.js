import { render } from "@testing-library/react";
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { TodoList } from "../../../components/08-useReducer/TodoList";
import { demoTodos } from "../../fixtures/demoTodo";


describe('Pruebas en <TodoList/>', () => {


    const handleDelete = jest.fn();
    const handleToggle = jest.fn();

    const wrapper = shallow(
        <TodoList
            todos={demoTodos}
            handleDelete={handleDelete}
            handleToggle={handleToggle}
        />
    )

    test('debe de mostrarse correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('debe de tener dos <TodoListItem/>', () => {
        expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);

        expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual(expect.any(Function));

    });


});
