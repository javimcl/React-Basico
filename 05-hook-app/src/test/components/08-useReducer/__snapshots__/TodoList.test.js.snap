// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoList/> debe de mostrarse correctamente 1`] = `
<div>
  <ui
    className="list-group list-group-flush"
  >
    <TodoListItem
      handleDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      i={0}
      key="1"
      todo={
        Object {
          "desc": "Aprender React",
          "done": false,
          "id": 1,
        }
      }
    />
    <TodoListItem
      handleDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      i={1}
      key="2"
      todo={
        Object {
          "desc": "Aprender Mongo",
          "done": false,
          "id": 2,
        }
      }
    />
  </ui>
</div>
`;
